### DISCUSSION TITLE
[Question]: Can the MemberCloner be used within a module?

### DISCUSSION THREAD

**Discussion by ds5678** [Washi1337/AsmResolver#288] [General]
For example, imagine I wanted to duplicate a `TypeDefinition` within a module and make some minor modifications. Would the `MemberCloner` let me do that, or would the references get scuffed?  This is unrelated to #281 (I use AsmResolver for multiple projects).

Related: #93

**Comment by Washi1337** [comment]
This should be fine. The MemberCloner is intended to be used for making deep copies of members that are fully imported into the target module. It heavily relies on the ReferenceImporter to do so. As such, if references come from a different module and are _not_ included in the cloner, it should always create _new_ definition and reference objects for these. If the target module is the same as the source module, then many definitions and references will be reused.
